submissions/Nemooochka/js-dom/js/script.js

```diff
@@ -0,0 +1,155 @@
+        'stars': 'Mark Ruffalo, Michael Keaton, Rachel McAdams'
+    },
+    {
+        'year': 'y_2014',
+        'name': 'Birdman',
+        'poster': 'https://m.media-amazon.com/images/M/MV5BODAzNDMxMzAxOV5BMl5BanBnXkFtZTgwMDMxMjA4MjE@._V1_UX182_CR0,0,182,268_AL_.jpg',
+        'description': 'A washed-up superhero actor attempts to revive his fading career by writing, directing, and starring in a Broadway production.',
+        'director': 'Alejandro G. Iñárritu',
+        'writers': 'Alejandro G. Iñárritu, Nicolás Giacobone',
+        'stars': 'Michael Keaton, Zach Galifianakis, Edward Norton'
+    }
+];
+
+const initialMovie = data[0];
+const navLink = document.querySelectorAll('.sidebar a');
+const sidebar = document.querySelector(".sidebar");
+const menuChevron = document.querySelector(".menu-chevron");
+
+const toggleSidebar = () => {
+    sidebar.classList.toggle('open');
+};
+
+/* get a year (last part of url) from href */
+const getLastItem = thePath => thePath.substring(thePath.lastIndexOf('/') + 1);
+
+function generateView(obj) {
+    return `<div class="card-poster">
+                <img src="${obj.poster}" alt="${obj.name}">
+            </div>
+            <div class="card-info">
+                <h2>${obj.name}</h2>
+                <div class="details">
+                    <p class="details_description">${obj.description}</p>
+                    <div class="details_director">
+                        <b>Director:</b>
+                        <span class="director">${obj.director}</span>
+                    </div>
+                    <div class="details_writer">
+                        <b>Writers:</b>
+                        <span>${obj.writers}</span>
+                    </div>
+                    <div class="details_stars">
+                        <b>Stars:</b>
+                        <span>${obj.stars}</span>
+                    </div>
+                </div>
+            </div>`;
+}
+
+function renderCard(nodeList) {
+    const cardMovie = document.querySelector('.card');
+    cardMovie.innerHTML = generateView(nodeList);
+}
+
+function updateActiveNav(targetLink) {
+    const getSiblings = el => {
+        let siblings = [];
+
+        // first child of the parent node
+        let sibling = el.parentNode.parentNode.firstChild;
+
+        // collecting siblings
+        while (sibling) {
+            if (sibling.nodeType === 1 && sibling !== el) {
+                siblings.push(sibling);
+            }
+            sibling = sibling.nextSibling;
+        }
+        return siblings;
+    };
+
+    const setActive = el => {
+        let nodes = getSiblings(el);
+        nodes.forEach(sib => sib.classList.remove('active'));
+        el.parentNode.classList.add('active')
+    };
+
+
+    setActive(targetLink);
+}
+
+function changeMovie(e) {
+    e.preventDefault();
+
+    updateActiveNav(this);
+
+    let chosenYear = getLastItem(this.href);
+    let chosenMovie;
+
+    for (let i = 0; i < data.length; i++) {
```

Comment:
It's better to use `Array.prototype.find`. More info [here](https://developer.mozilla.org/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/find)